src/platform/test/functional/apps/discover/esql/_esql_columns.ts

```diff
@@ -275,9 +275,11 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
       await browser.refresh();
       await discover.waitUntilTabIsLoaded();
       await timePicker.setDefaultAbsoluteRange();
-      await testSubjects.click('querySubmitButton');
-      await discover.waitUntilTabIsLoaded();
-      expect(await dataGrid.getHeaderFields()).to.eql(['ip', '@timestamp']);
+      await retry.try(async () => {
+        await testSubjects.click('querySubmitButton');
+        await discover.waitUntilTabIsLoaded();
+        expect(await dataGrid.getHeaderFields()).to.eql(['ip', '@timestamp']);
+      });
```

Comment:
I think the test is checking that columns are correct after a browser refresh. Adding `await testSubjects.click('querySubmitButton');` would contradict with the test. 

Can we try something like this instead?

```suggestion
      await discover.waitUntilTabIsLoaded();
      await retry.try(async () => {
        expect(await dataGrid.getHeaderFields()).to.eql(['ip', '@timestamp']);
      });
```

Reply:
thx, I was looking into that, and we might further simplify, it could be the case that the flakiness was coming from the fact that we try submit the  query twice, first when changing the time range, then explicitly again, will test another approach here 👍